pkg/util/testutil/config_change_checker.go

```diff
@@ -0,0 +1,58 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package testutil
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+// ConfigChangeChecker checks the state of `config.Datadog` did not change
+// between `NewConfigChangeChecker()`` and `HasChanged()`. It is
+// designed to be used in `TestMain` function as follow:
+//
+// func TestMain(m *testing.M) {
+// 	checker := testutil.NewConfigChangeChecker()
+// 	exit := m.Run()
+// 	if checker.HasChanged() {
+// 		os.Exit(1)
+// 	}
+// 	os.Exit(exit)
+// }
```

Comment:
To avoid wrapping in the generated docs, you'll want to indent this example code.